lib/request_handler/multipart_parser.rb

```diff
@@ -35,13 +37,19 @@ def parse_part(name)
     end
 
     def parse_data(name)
-      JsonApiDocumentParser.new(
-        document:         load_json(name),
+      data = load_json(name)
+      parser = jsonapi?(name) ? JsonApiDocumentParser : JsonParser
+      parser.new(
+        document:         data,
         schema:           lookup("#{name}.schema"),
         schema_options:   execute_options(lookup("#{name}.options"))
       ).run
     end
 
+    def jsonapi?(name)
+      request.params[name][:type] == 'application/vnd.api+json'
```

Comment:
please deduplicate this into a factory which you use both here and in the normal request body handler